include/dialogs/PreviewImageOverlay.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * nheko Copyright (C) 2017  Konstantinos Sideris <siderisk@auth.gr>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <QLabel>
+#include <QLineEdit>
+#include <QPixmap>
+#include <QWidget>
+
+#include "FlatButton.h"
+
+class QMimeData;
+
+namespace dialogs {
+
+class PreviewImageOverlay : public QWidget
+{
+        Q_OBJECT
+public:
+        PreviewImageOverlay(QWidget *parent = nullptr);
+
+        void setImageAndCreate(const QByteArray &data, const QString &type);
```

Comment:
To avoid inadvertent sharing and possible UB, this should take a `QByteArray` by value. [`QByteArray` is internally ref-counted](https://doc.qt.io/qt-5/qbytearray.html#QByteArray-4) so doing so is cheap.